examples/ErrorHandling/HandlePartialFailures.cs

```diff
@@ -0,0 +1,186 @@
+        {
+            // Get the AdGroupCriterionService.
+            AdGroupCriterionServiceClient adGroupCriterionService =
+                client.GetService(Services.V1.AdGroupCriterionService);
+
+            // Create the keywords. The second and fourth keywords have policy violation, and will
+            // fail with errors.
+            string[] keywords = new string[]
+            {
+                "mars cruise",
+                "inv@lid cruise",
+                "venus cruise",
+                "b(a)d keyword cruise"
+            };
+
+            List<AdGroupCriterionOperation> operations = new List<AdGroupCriterionOperation>();
+            for (int i = 0; i < keywords.Length; i++)
+            {
+                // Create a keyword.
+                AdGroupCriterion criterion = new AdGroupCriterion()
+                {
+                    AdGroup = ResourceNames.AdGroup(customerId, adGroupId),
+                    Status = AdGroupCriterionStatus.Enabled,
+                    Keyword = new KeywordInfo()
+                    {
+                        Text = keywords[i],
+                        MatchType = KeywordMatchType.Exact
+                    }
+                };
+
+                // Create the operation.
+                AdGroupCriterionOperation operation = new AdGroupCriterionOperation()
+                {
+                    Create = criterion,
+                };
+                operations.Add(operation);
+            }
+
+            try
+            {
+                // Add the keywords.
+                MutateAdGroupCriteriaResponse result =
+                    adGroupCriterionService.MutateAdGroupCriteria(customerId.ToString(),
+                        operations, true, false);
+
+                // Display the results.
+                if (result.Results.Count > 0)
+                {
+                    foreach (MutateAdGroupCriterionResult newCriterion in result.Results)
+                    {
+                        // If an operation failed, the result would be empty.
+                        if (!newCriterion.IsEmpty())
+                        {
+                            Console.WriteLine($"Created keyword with resource ID = " +
+                                $"'{newCriterion.ResourceName}'.");
+                        }
+                    }
+                }
+
+                // Display the partial failure errors.
+                foreach (GoogleAdsFailure failure in result.PartialFailures)
+                {
+                    foreach (GoogleAdsError error in failure.Errors)
+                    {
+                        List<string> fieldPaths = new List<string>();
+                        long? operationIndex = null;
+                        foreach (FieldPathElement fpElement in error.Location.FieldPathElements)
+                        {
+                            if (fpElement.FieldName == "operation")
```

Comment:
I think you could use some comments here explaininig what this does and how a user would be expected to interact with these objects. For java, I added this to a utility class: https://github.com/googleads/google-ads-java/blob/master/google-ads/src/main/java/com/google/ads/googleads/v1/utils/ErrorUtils.java